pytest-server-fixtures/pytest_server_fixtures/s3.py

```diff
@@ -0,0 +1,94 @@
+# coding: utf-8
+"""
+Pytest fixtures to launch a minio S3 server and get a bucket for it.
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import uuid
+
+import boto3
+import botocore.client
+import pytest
+from future.utils import text_type
+from pytest_fixture_config import requires_config
+
+from . import CONFIG
+from .base import TestServer
+
+
+def _s3_server(request):
+    server = MinioServer()
+    server.start()
+    request.addfinalizer(server.teardown)
+    return server
+
+
+@requires_config(CONFIG, ['minio_bin'])
+@pytest.fixture(scope="session")
+def s3_server(request):
```

Comment:
Can we add a docstring here, as this will show up in the help text when people run 'py.test --fixtures'. Have a look at some of the other fixtures for examples - at a minimum just let people know what they get back in the fixture and what sort of attributes it might have.